### PR TITLE
Port over the new StringStream and net streaming functionality from starfall

### PR DESCRIPTION
**Pull request by CapsAdmin**
This is not 100% yet, there's currently an infinite loop in mdlzip functionality.

### CODE REVIEW

File: lua/pac3/core/shared/util.lua [hunk 1]
```diff
@@ -647,37 +647,43 @@ function pac.DownloadMDL(url, callback, onfail, ply)
...
 
-						-- Add nulls to align to 4 bytes
-						while (#f.buffer)%4 ~= 0 do
-							f.buffer[#f.buffer+1] = 0
+
+						do
+							error("NYI, code below infinite loops")
+
+							-- Add nulls to align to 4 bytes
+							while (#f)%4 ~= 0 do
```
Comment (CapsAdmin) [inline lua/pac3/core/shared/util.lua:670]: have a look here @thegrb93

**Comment by thegrb93** [comment]
I'll correct it. The new version has some differences. Also Seek and Tell are 1 first index like lua instead of 0 first index.

**Comment by thegrb93** [comment]
Works with my outfit. 👍

**Comment by thegrb93** [comment]
I think that's everything now. Might want @Yagira to also test that it still works.

**Comment by Yagira** [comment]
> Might want @Yagira to also test that it still works.

All my models seem to work fine.

**Comment by CapsAdmin** [comment]
I think that's good enough for the develop branch.

**Comment by thegrb93** [comment]
@CapsAdmin One thing I just remembered. The reason to make net_stream global is so that it is shared between starfall/advdupe2/pac3. That way two won't be running concurrently and potentially overflow the net buffer.

**Comment by thegrb93** [comment]
It might be a good idea to switch it to be global again.

**Pull request merged**